### PR TITLE
Update 'adding your own workflow' documentation

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-1650](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1650)

### CODE REVIEW

File: docs/workflows/developing-a-cumulus-workflow.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+hide_title: true
+---
+
+# Creating a Cumulus workflow
+
+## The Cumulus workflow module
+
+To facilitate adding a workflows to your deployment Cumulus provides a [workflow module](https://github.com/nasa/cumulus/tree/master/tf-modules/workflow).
+
+In combination with the [Cumulus message](cumulus-task-message-flow) it provides a way to easily turn a step function definition into a Cumulus workflow, complete with [AWS CloudWatch events](https://docs.aws.amazon.com/cloudwatch/index.html) that handle database updates for Cumulus objects (e.g. `executions`, `granules`, etc), can seamlessly take advantage of of our [throttling](../data-cookbookx/throttling-queued-executions) feature, as well as ensuring that your workflows will continue to be compatible with future versions of Cumulus.
```
[the remark this answers is not in the document]
Comment (markdboyd) [reply docs/workflows/developing-a-cumulus-workflow.md]: As is the link for throttling (`data-cookbooks`, not `data-cookbookx`)

[the remark this answers is not in the document]
Comment (markdboyd) [reply docs/workflows/developing-a-cumulus-workflow.md]:
```suggestion
In combination with the [Cumulus message](cumulus-task-message-flow), the workflow module provides a way to easily turn a Step Function definition into a Cumulus workflow, complete with:

- [AWS CloudWatch events](https://docs.aws.amazon.com/cloudwatch/index.html) that handle database updates for Cumulus objects (e.g. `executions`, `granules`, etc)
- Built-in integration with our [throttling](../data-cookbookx/throttling-queued-executions) feature

Using the module also ensures that your workflows will continue to be compatible with future versions of Cumulus.
```

[the remark this answers is not in the document]
Comment (markdboyd) [reply docs/workflows/developing-a-cumulus-workflow.md]: What about using the workflow module gives integration with the throttling feature?

File: docs/workflows/developing-a-cumulus-workflow.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  state_machine_definition = <<JSON
+{}
+JSON
+}
+```
+
+In the above example, you would add your `state_machine_definition` using the [Amazon States Language](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html), using [tasks you've developed](developing-workflow-tasks) and [Cumulus core tasks](../tasks) that are made available as part of the `cumulus` terraform module.
+
+**Please note**: Cumulus follows the convention of tagging resources with the `prefix` variable `{ Deployment = var.prefix }` that you pass to the `cumulus` module.   For resources defined outside of Core, it's recommended that you adopt this convention as it makes resources and/or deployment recovery scenarios much easier to manage.
```
Comment (markdboyd) [inline docs/workflows/developing-a-cumulus-workflow.md]: 👍

File: docs/workflows/developing-a-cumulus-workflow.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  tags = { Deployment = var.prefix }
+
+  state_machine_definition = <<JSON
+{}
+JSON
+}
+```
+
+In the above example, you would add your `state_machine_definition` using the [Amazon States Language](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html), using [tasks you've developed](developing-workflow-tasks) and [Cumulus core tasks](../tasks) that are made available as part of the `cumulus` terraform module.
```
Comment (markdboyd) [inline docs/workflows/developing-a-cumulus-workflow.md]: Is this `developing-workflow-tasks` link working?

File: docs/workflows/developing-a-cumulus-workflow.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+**Please note**: Cumulus follows the convention of tagging resources with the `prefix` variable `{ Deployment = var.prefix }` that you pass to the `cumulus` module.   For resources defined outside of Core, it's recommended that you adopt this convention as it makes resources and/or deployment recovery scenarios much easier to manage.
+
+## Examples
+
+For a functional example of a basic workflow, please take a look at the [hello_world_workflow](https://github.com/nasa/cumulus-template-deploy/blob/CUMULUS-1556-add-deployment/cumulus-tf/hello_world_workflow.tf).
```
Comment (markdboyd) [inline docs/workflows/developing-a-cumulus-workflow.md]: Need to update the ref here to `master`

File: docs/workflows/developing-a-cumulus-workflow.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+For  more complete/advanced examples, please read the following cookbook entries/topics:
```
Comment (markdboyd) [inline docs/workflows/developing-a-cumulus-workflow.md]:
```suggestion
For more complete/advanced examples, please read the following cookbook entries/topics:
```

File: docs/workflows/lambda.md [hunk 6]
```diff
@@ -67,3 +69,15 @@ For Lambdas wishing to utilize the [Cumulus Message Adapter(CMA)](cumulus-task-m
 ## Other Lambda Options
 
 Cumulus supports all of the options available to you via the `aws_lambda_function` Terraform resource.   For more information on what's available, check out the [Terraform resource docs](https://www.terraform.io/docs/providers/aws/r/lambda_function.html).
+
+### Cloudwatch log groups
+
+If you want to enable [Cloudwatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/index.html) logging for your Lambda resource, you'll need to add a `aws_cloudwatch_log_group` resource to your Lambda definition:
+
+```hcl
+resource "aws_cloudwatch_log_group" "sync_granule_task" {
```
Comment (markdboyd) [inline docs/workflows/lambda.md]:
```suggestion
resource "aws_cloudwatch_log_group" "myfunction_log_group" {
```

File: docs/workflows/developing-a-cumulus-workflow.md (hunk 4 above)
Comment (Jkovarik) [reply docs/workflows/developing-a-cumulus-workflow.md]: c3c951ee4

File: docs/workflows/developing-a-cumulus-workflow.md (hunk 3 above)
Comment (Jkovarik) [reply docs/workflows/developing-a-cumulus-workflow.md]: I just fired it up locally to re-check and it appears to be, yes.

**Comment by Jkovarik** [comment]
@markdboyd  thanks for the review, I updated the link with the bad ref, I think everything else is sorted.

File: docs/workflows/developing-a-cumulus-workflow.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+# Creating a Cumulus workflow
+
+## The Cumulus workflow module
+
+To facilitate adding a workflows to your deployment Cumulus provides a [workflow module](https://github.com/nasa/cumulus/tree/master/tf-modules/workflow).
+
+In combination with the [Cumulus message](cumulus-task-message-flow), the workflow module provides a way to easily turn a Step Function definition into a Cumulus workflow, complete with:
+
+- [AWS CloudWatch events](https://docs.aws.amazon.com/cloudwatch/index.html) that handle database updates for Cumulus objects (e.g. `executions`, `granules`, etc)
+- Built-in integration with our [throttling](../data-cookbookx/throttling-queued-executions) feature
```
Comment (laurenfrederick) [inline docs/workflows/developing-a-cumulus-workflow.md]: This link isn't working for me - you have an `x` in there

File: docs/workflows/developing-a-cumulus-workflow.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+Using the module also ensures that your workflows will continue to be compatible with future versions of Cumulus.
+
+For more on the full set of current available options for the module, please consult the [README](https://github.com/nasa/cumulus/blob/master/tf-modules/workflow/README.md).
```
Comment (laurenfrederick) [inline docs/workflows/developing-a-cumulus-workflow.md]:
```suggestion
For more on the full set of current available options for the module, please consult module the [README](https://github.com/nasa/cumulus/blob/master/tf-modules/workflow/README.md).
```

So nitpicky

File: docs/workflows/developing-a-cumulus-workflow.md (hunk 4 above)
Comment (markdboyd) [reply docs/workflows/developing-a-cumulus-workflow.md]: Hmm,  the link here still doesn't point to the `master` branch?

Comment (Jkovarik) [reply docs/workflows/developing-a-cumulus-workflow.md]: It would if my push succeed.

Comment (Jkovarik) [reply docs/workflows/developing-a-cumulus-workflow.md]: c3c951ee4

File: docs/workflows/developing-a-cumulus-workflow.md (hunk 8 above)
Comment (Jkovarik) [reply docs/workflows/developing-a-cumulus-workflow.md]: ` please consult module the` ?

Comment (Jkovarik) [inline docs/workflows/developing-a-cumulus-workflow.md]: How about: 

```suggestion
For more on the full set of current available options for the module, please consult the module [README](https://github.com/nasa/cumulus/blob/master/tf-modules/workflow/README.md).
```

File: docs/workflows/developing-a-cumulus-workflow.md (hunk 7 above)
Comment (Jkovarik) [reply docs/workflows/developing-a-cumulus-workflow.md]: 4fd927328

**Pull request merged**